stdlib/LibGit2/src/LibGit2.jl

```diff
@@ -12,7 +12,7 @@ using Base.Printf: @printf
 export with, GitRepo, GitConfig
 
 const GITHUB_REGEX =
-    r"^(?:git@|git://|https://(?:[\w\.\+\-]+@)?)github.com[:/](([^/].+)/(.+?))(?:\.git)?$"i
+    r"^(?:git@|git://|https://|ssh://git@(?:[\w\.\+\-]+@)?)github.com[:/](([^/].+)/(.+?))(?:\.git)?$"i
```

Comment:
This change breaks HTTPS as the general username parsing needs to be applied only on that protocol. A test should be added along the lines of: "https://username@github.com"

Instead, we can do:
```suggestion
    r"^(?:(?:ssh://)?git@|git://|https://(?:[\w\.\+\-]+@)?)github.com[:/](([^/].+)/(.+?))(?:\.git)?$"i
```